velox/functions/prestosql/aggregates/tests/AggregationTestBase.cpp

```diff
@@ -32,4 +35,135 @@ std::vector<RowVectorPtr> AggregationTestBase::makeVectors(
+    const std::vector<std::string>& aggregates,
+    const std::string& duckDbSql) {
+  testAggregations(data, groupingKeys, aggregates, {}, duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    const std::vector<RowVectorPtr>& data,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& postAggregationProjections,
+    const std::string& duckDbSql) {
+  testAggregations(
+      [&](PlanBuilder& builder) { builder.values(data); },
+      groupingKeys,
+      aggregates,
+      postAggregationProjections,
+      duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    std::function<void(PlanBuilder&)> makeSource,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::string& duckDbSql) {
+  testAggregations(makeSource, groupingKeys, aggregates, {}, duckDbSql);
+}
+
+void AggregationTestBase::testAggregations(
+    std::function<void(PlanBuilder&)> makeSource,
+    const std::vector<std::string>& groupingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& postAggregationProjections,
+    const std::string& duckDbSql) {
+  // Run partial + final.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.partialAggregation(groupingKeys, aggregates).finalAggregation();
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run single.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.singleAggregation(groupingKeys, aggregates);
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run partial + intermediate + final.
+  {
+    PlanBuilder builder;
+    makeSource(builder);
+    builder.partialAggregation(groupingKeys, aggregates)
+        .intermediateAggregation()
+        .finalAggregation();
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+    assertQuery(builder.planNode(), duckDbSql);
+  }
+
+  // Run partial + local exchange + final.
+  if (!groupingKeys.empty()) {
+    auto planNodeIdGenerator =
+        std::make_shared<exec::test::PlanNodeIdGenerator>();
+    PlanBuilder sourceBuilder{planNodeIdGenerator};
+    makeSource(sourceBuilder);
+
+    auto builder =
+        PlanBuilder(planNodeIdGenerator)
+            .localPartition(
+                groupingKeys,
+                {sourceBuilder.partialAggregation(groupingKeys, aggregates)
+                     .planNode()})
+            .finalAggregation();
+    if (!postAggregationProjections.empty()) {
+      builder.project(postAggregationProjections);
+    }
+
+    CursorParameters params;
+    params.planNode = builder.planNode();
+    params.maxDrivers = 2;
```

Comment:
qq : Any reason the maxDrivers is hard-coded and not passed as a parameter ?

Reply:
The only reason is to simplify the API.